File: qqbot-runtime/src/main/java/com/mieai/qqbot/runtime/configuration/BotConfigurationView.java
package com.mieai.qqbot.runtime.configuration;

import com.mieai.qqbot.domain.bot.BotEnvironment;
import com.mieai.qqbot.domain.bot.BotDefinition;
import com.mieai.qqbot.domain.bot.BotId;
import com.mieai.qqbot.domain.bot.BotRevision;
import com.mieai.qqbot.domain.bot.GatewayIntents;
import com.mieai.qqbot.domain.bot.QqAppId;
import com.mieai.qqbot.domain.bot.ShardSpec;
import java.time.Instant;
import java.util.Objects;

/** Read model that deliberately exposes only whether an AppSecret is configured. */
public record BotConfigurationView(
        BotId id,
        String displayName,
        QqAppId appId,
        BotEnvironment environment,
        GatewayIntents intents,
        ShardSpec shardSpec,
        boolean enabled,
        BotRevision revision,
        Instant createdAt,
        Instant updatedAt,
        boolean secretConfigured,
        long maxMediaUploadBytes) {

    public BotConfigurationView(BotId id, String displayName, QqAppId appId,
            BotEnvironment environment, GatewayIntents intents, ShardSpec shardSpec,
            boolean enabled, BotRevision revision, Instant createdAt, Instant updatedAt,
            boolean secretConfigured) {
        this(id, displayName, appId, environment, intents, shardSpec, enabled, revision,
                createdAt, updatedAt, secretConfigured, BotDefinition.DEFAULT_MAX_MEDIA_UPLOAD_BYTES);
    }

    public BotConfigurationView {
        Objects.requireNonNull(id, "id must not be null");
        displayName = ConfigurationValidation.displayName(displayName);
        Objects.requireNonNull(appId, "appId must not be null");
        Objects.requireNonNull(environment, "environment must not be null");
        Objects.requireNonNull(intents, "intents must not be null");
        Objects.requireNonNull(shardSpec, "shardSpec must not be null");
        Objects.requireNonNull(revision, "revision must not be null");
        Objects.requireNonNull(createdAt, "createdAt must not be null");
        Objects.requireNonNull(updatedAt, "updatedAt must not be null");
        if (maxMediaUploadBytes < BotDefinition.MIN_MAX_MEDIA_UPLOAD_BYTES
                || maxMediaUploadBytes > BotDefinition.MAX_MAX_MEDIA_UPLOAD_BYTES) {
            throw new IllegalArgumentException("maxMediaUploadBytes must be between 1 MiB and 256 MiB");
        }
    }
}

File: qqbot-runtime/src/main/kotlin/com/mieai/qqbot/runtime/configuration/BotConfigurationView.kt
package com.mieai.qqbot.runtime.configuration

import com.mieai.qqbot.domain.bot.*
import java.time.Instant

data class BotConfigurationView(val id: BotId, val displayName: String, val appId: QqAppId, val environment: BotEnvironment, val intents: GatewayIntents, val shardSpec: ShardSpec, val enabled: Boolean, val revision: BotRevision, val createdAt: Instant, val updatedAt: Instant, val secretConfigured: Boolean, val maxMediaUploadBytes: Long = BotDefinition.DEFAULT_MAX_MEDIA_UPLOAD_BYTES) {
    init {
        ConfigurationValidation.displayName(displayName)
        require(maxMediaUploadBytes in BotDefinition.MIN_MAX_MEDIA_UPLOAD_BYTES..BotDefinition.MAX_MAX_MEDIA_UPLOAD_BYTES) { "maxMediaUploadBytes must be between 1 MiB and 256 MiB" }
    }
}
